docs/markdown/Reference-manual.md

```diff
@@ -2724,6 +2724,24 @@ The following methods are defined for all [`feature` options](Build-options.md#f
 - `enabled()`: returns whether the feature was set to `'enabled'`
 - `disabled()`: returns whether the feature was set to `'disabled'`
 - `auto()`: returns whether the feature was set to `'auto'`
+- `allowed()` *(since 0.58.0)*: returns whether the feature was set to `'enabled'` or `'auto'`
+- `may_disable_if(value)` *(since 0.58.0)*: returns the feature, with
+  `'auto'` converted to `'disabled'` if value is true.
```

Comment:
If I understand correctly, the use case is something like:
`directx_opt = get_option('directx').may_disable_if(host_machine.os != 'windows')`

Could be just me, but I find the name a bit weird. What about shorter `.disabled_if()`?

Reply:
Now that I have a better understanding of the use-case for this, I like it... But I think we should include example(s) in the documentation to help understanding the intended usage. Since you're using directx story for `.require()`, would make sense to include my example from https://github.com/mesonbuild/meson/pull/8512#issuecomment-799612560.

Reply:
What about also renaming `value` to `condition` here for clarity?

Reply:
What about even adding a table in the doc to make it totally clear:

| Feature / Condition  | True | False |
| -------------------------- | ------ | ------- |
| Enabled                   | Enabled | Enabled |
| Disabled                  | Disabled | Disabled |
| Auto                         | Disabled | Auto |